bot.py

```diff
@@ -175,8 +185,113 @@ async def author_counts() -> None:
+        
+        """Creates sets from the submitter names and their counts """
+        
+        author_set = [([(author), (count)]) for author, count in authors.items()]
+    
+        """Determines if the graph is linear or logarithmic, then bar or line
+        then calls a function to generate the graph"""
+
+        is_linear = bool
+        if which_scale == "linear":
+            is_linear = True
+        elif which_scale == "log":
+            is_linear = False
+        else:
+            logger.info("An invalid scale type was requested!")
+            embed_msg = discord.Embed(title="Error", colour=random_colour(), description="error message")
+            embed_msg.set_footer(text="An invalid scale type has been requested!")
+
+            await client.get_channel(CHANNEL).send(embed=embed_msg)
+            return
+        
+        if which_graph == "line":
+            await send_graph(author_set, False, is_linear)
+        elif which_graph == "bar":
+            await send_graph(author_set, True, is_linear)
+        else:
+            logger.info("An invalid graph type was requested!")
+            embed_msg = discord.Embed(title="Error", colour=random_colour(), description="error message")
+            embed_msg.set_footer(text="An invalid graph type has been requested!")
+
+            await client.get_channel(CHANNEL).send(embed=embed_msg)
+        
 
+@client.event
+async def send_graph(author_set: list[tuple[str, int]], is_bar: bool, is_linear: bool) -> None:
+
+        """Define the scaling of the graph"""
+    
+        x_size = 0.5 * len(author_set)
+        y_size = 0.65* x_size
+        fig_size = plt.rcParams["figure.figsize"]
+        fig_size[0] = x_size
+        fig_size[1] = y_size
+        plt.rcParams["figure.figsize"] = fig_size
+        plt.rcParams.update({"font.size": 14})
+
+        """Split these into data points"""
+        
+        x = [plot_point[0] for plot_point in author_set]
+        y = [plot_point[1] for plot_point in author_set]
+
+
+        """Plot either a bar or line, and annotate each one appropriately"""
+        if is_bar:
+            plt.bar(x,y)
+            for i, num in enumerate(y):
+                plt.annotate(num, (x[i], y[i]), xycoords ="data", xytext = (-4, 3), textcoords="offset points")
+
+        else:
+            plt.scatter(x,y)
+            plt.plot(x,y)
+            for i, num in enumerate(y):
+                if i > 0:
+                    plt.annotate(num, (x[i], y[i]), xycoords ="data", xytext = (4, 4), textcoords="offset points")
+            plt.annotate(y[0], (x[0], y[0]), xycoords ="data", xytext = (4, -2), textcoords="offset points")
+
+
+
+        """Make the graph look pretty"""
+        
+        plt.xticks(rotation=45, ha="right")
+        plt.xlabel ("Name of Submitters")
+        plt.ylabel ("Number of Submissions")
+        plt.title ("Chart of Submissions to SwackQuote!")
+
+
+        """Determine the scale of the graph"""
+        
+        if is_linear:
```

Comment:
Since we already know the type will be correct, and this is a boolean, we can replace this with a ternary conditional to clean up the code; something like `plt.yscale("linear" if is_linear else "log")

Reply:
We also aren't really determining the type here, more so setting it, but that's a comment